### PR TITLE
feat(ext/web): add performance.toJSON

### LINKED ISSUE #10838: Missing performance.toJSON

### PR DESCRIPTION
**Pull request by GJZwiers**
Closes #10838. Needs https://github.com/denoland/deno_std/pull/2209

### CODE REVIEW

File: cli/tests/unit/performance_test.ts [hunk 1]
```diff
@@ -27,6 +27,12 @@ Deno.test(function timeOrigin() {
   assert(Date.now() >= origin);
 });
 
+Deno.test(function performanceToJSON() {
+  const json = performance.toJSON();
+
+  assert("timeOrigin" in json);
```
Comment (cjihrig) [inline cli/tests/unit/performance_test.ts:33]: It might be useful to also assert on:

1. There are no other keys.
2. The data type of `timeOrigin`.

Comment (GJZwiers) [reply cli/tests/unit/performance_test.ts:33]: I have added more assertions!

File: cli/tests/unit/performance_test.ts [hunk 2]
```diff
@@ -27,6 +27,15 @@ Deno.test(function timeOrigin() {
...
+  assert(typeof json.timeOrigin === "number");
```
Comment (cjihrig) [inline cli/tests/unit/performance_test.ts:34]: Sorry for not saying this in my original comment: couldn't this be `assert(json.timeOrigin === performance.timeOrigin);`

**Comment by bartlomieju** [comment]
Looks like tests are failing on incompatible typing in `deno_std/node`. We probably need to update the typings there and update the git submodule before we can land this PR. @GJZwiers could you open a PR to `deno_std` to fix this?

**Comment by GJZwiers** [comment]
> Looks like tests are failing on incompatible typing in `deno_std/node`. We probably need to update the typings there and update the git submodule before we can land this PR. @GJZwiers could you open a PR to `deno_std` to fix this?

Right now the changes in `std` won't pass the type-checking, because they need the updated `.d.ts` from this PR.

**Comment by bartlomieju** [comment]
Yikes this is unlucky... we got a chicken and egg problem here. Please open a PR in deno_std anyway, I guess we'll disable these tests for a moment in CLI and then upgrade `deno_std` submodule. Unfortunately it might have to wait a week until 1.22 release to land it.

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @GJZwiers

**Pull request merged**